pandas/core/window.py

```diff
@@ -0,0 +1,1339 @@
+                results[i] = f(*_prep_binary(arg1.iloc[:, i], arg2))
+            return dataframe_from_int_dict(results, arg1)
+
+    else:
+        return _flex_binary_moment(arg2, arg1, f)
+
+def _get_center_of_mass(com, span, halflife):
+    valid_count = len([x for x in [com, span, halflife] if x is not None])
+    if valid_count > 1:
+        raise Exception("com, span, and halflife are mutually exclusive")
+
+    if span is not None:
+        # convert span to center of mass
+        com = (span - 1) / 2.
+    elif halflife is not None:
+        # convert halflife to center of mass
+        decay = 1 - np.exp(np.log(0.5) / halflife)
+        com = 1 / decay - 1
+    elif com is None:
+        raise Exception("Must pass one of com, span, or halflife")
+
+    return float(com)
+
+def _offset(window, center):
+    if not com.is_integer(window):
+        window = len(window)
+    offset = (window - 1) / 2. if center else 0
+    try:
+        return int(offset)
+    except:
+        return offset.astype(int)
+
+def _require_min_periods(p):
+    def _check_func(minp, window):
+        if minp is None:
+            return window
+        else:
+            return max(p, minp)
+    return _check_func
+
+def _use_window(minp, window):
+    if minp is None:
+        return window
+    else:
+        return minp
+
+def _zsqrt(x):
+    result = np.sqrt(x)
+    mask = x < 0
+
+    from pandas import DataFrame
+    if isinstance(x, DataFrame):
+        if mask.values.any():
+            result[mask] = 0
+    else:
+        if mask.any():
+            result[mask] = 0
+
+    return result
+
+def _prep_binary(arg1, arg2):
+    if not isinstance(arg2, type(arg1)):
+        raise Exception('Input arrays must be of the same type!')
+
+    # mask out values, this also makes a common index...
+    X = arg1 + 0 * arg2
+    Y = arg2 + 0 * arg1
+
+    return X, Y
+
+def _validate_win_type(win_type, kwargs):
+    # may pop from kwargs
+    arg_map = {'kaiser': ['beta'],
+               'gaussian': ['std'],
+               'general_gaussian': ['power', 'width'],
+               'slepian': ['width']}
+    if win_type in arg_map:
+        return tuple([win_type] +
+                     _pop_args(win_type, arg_map[win_type], kwargs))
+    return win_type
+
+
+def _pop_args(win_type, arg_names, kwargs):
+    msg = '%s window requires %%s' % win_type
+    all_args = []
+    for n in arg_names:
+        if n not in kwargs:
+            raise ValueError(msg % n)
+        all_args.append(kwargs.pop(n))
+    return all_args
+
+#############################
+##### top-level exports #####
+#############################
+
+def rolling(obj, win_type=None, **kwds):
+    from pandas import Series, DataFrame
+    if not isinstance(obj, (Series, DataFrame)):
+        raise TypeError('invalid type: %s' % type(obj))
+
+    if win_type is not None:
+        return Window(obj, win_type=win_type, **kwds)
+
+    return Rolling(obj, **kwds)
+rolling.__doc__ = Window.__doc__
```

Comment:
ah, yes, that is certainly fine! It was also because I noticed that the `DataFrame.rolling` had no mention of `freq` being deprecated (which came from Window.__doc__), but I already commented for that on the actual docstring